tests/Yii/Comment.php

```diff
@@ -18,6 +18,14 @@ public static function find(): CommentsQuery {
         return new CommentsQuery(self::class);
     }
 
+    /**
+     * @param string $sql
+     * @param array<string, mixed> $params
+     */
+    public static function findBySql($sql, $params = []): CommentsQuery {
+        return new CommentsQuery(self::class);
+    }
```

Comment:
According to the [implementation inside of Yii2](https://github.com/yiisoft/yii2/blob/997559760638557140169f01e8c2514263a7f986/framework/db/ActiveRecord.php#L163-L164) it must inherit return type when the `find()` method is overridden. It would be great if you could do that as part of this PR. If you don't know how to do it, you can safely skip this comment and I will do it myself later.

Reply:
I'd like to, but i'm not shure what exactly you're talking about here.
Do you mean that we should be able to call
`Comment::findBySql('')->notDeletedSelf()->all()`
without overriding `findBySql` in the `Comment.php` and get expected types and results since we have `find` overridden?

Reply:
Yes, exactly. `::findBySql()`, when not overridden manually, must inherit the return type from `::find() `method.

Reply:
I kinda did it, but i'm not sure you will like the way i did it.

Reply:
Unfortunately, no, the suggested solution is kinda weird. It would be simpler if I'll finish this part on my own (if you don't mind).